### ISSUE TITLE
Error when merging two Yi models

### ISSUE THREAD

**Issue by mlinmg** [cg123/mergekit#12]
Hi, awesome project
Anyway I'm encountering majour difficulties when trying to fuse two Yi models([1](https://huggingface.co/NousResearch/Nous-Capybara-34B),[2](https://huggingface.co/migtissera/Tess-M-v1.3/tree/main))
this its the conf file 
```
slices:
  - sources:
      - model: /workspace/Tess-M-v1.3
        layer_range: [0, 14]
  - sources:
      - model: /workspace/Nous-Capybara-34B
        layer_range: [7, 21]  
  - sources:
      - model: /workspace/Tess-M-v1.3
        layer_range: [15, 29] 
  - sources:
      - model: /workspace/Nous-Capybara-34B
        layer_range: [22, 36] 
  - sources:
      - model: /workspace/Tess-M-v1.3
        layer_range: [30, 44] 
  - sources:
      - model: /workspace/Nous-Capybara-34B
        layer_range: [37, 51]  
  - sources:
      - model: /workspace/Tess-M-v1.3
        layer_range: [45, 59] 
merge_method: passthrough
base_model: /workspace/Tess-M-v1.3
tokenizer_source: /workspace/Tess-M-v1.3
dtype: float16
```
It doesn't seem to change anything if I switch the models.
The error it give is:
```
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
Traceback (most recent call last):

  File "/usr/local/bin/mergekit-yaml", line 8, in <module>
    sys.exit(_main())

  File "/workspace/mergekit/mergekit/scripts/run_yaml.py", line 101, in _main
    typer.run(main)

  File "/workspace/mergekit/mergekit/scripts/run_yaml.py", line 82, in main
    run_merge(

  File "/workspace/mergekit/mergekit/merge.py", line 68, in run_merge
    tokenizer, embed_permutations = build_tokenizer(

  File "/workspace/mergekit/mergekit/tokenizer.py", line 54, in build_tokenizer
    vocabularies[model] = model_tok.vocab

AttributeError: 'YiTokenizer' object has no attribute 'vocab'
```

**Comment by mlinmg** [comment]
If I remove this line it works, but it generate a model which throws this error  Trying to set a tensor of shape torch.Size([64000, 7168]) in "weight" (which has shape torch.Size([64002, 7168])), this look incorrect.,

**Comment by cg123** [comment]
Looks like you've found a bug in my tokenizer merging code. I've fixed that error in 466f301ea916677de7429a35230b1d81befaa4d8 - YiTokenizer should work fine now. If it still gives you the tensor size error let me know and I can try to hunt it down.

**Comment by brucethemoose** [comment]
For what its worth, those to models successfully merged before. Perhaps because it was using a llamafied Yi as a base model?
